lib/src/component_config/components/config_parallel_prefix_adder.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// config_parallel-prefix_adder.dart
+// Configurator for a Parallel Prefix Adder.
+//
+// 2024 February 5
+// Author: Desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:collection';
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// A [Configurator] for [ParallelPrefixAdder]s.
+class ParallelPrefixAdderConfigurator extends Configurator {
+  /// Controls the type of [ParallelPrefix] tree used in the adder.
+  final ChoiceConfigKnob<
+          ParallelPrefix Function(List<Logic>, Logic Function(Logic, Logic))>
+      prefixTreeKnob = ChoiceConfigKnob(
+          [Ripple.new, Sklansky.new, KoggeStone.new, BrentKung.new],
```

Comment:
Try running `toJson` on these and check that the values don't have some gobbledygook pointer reference or something as the name?

Reply:
```
void main() {
  final prefixTreeKnob = ChoiceConfigKnob(
      [Ripple.new, Sklansky.new, KoggeStone.new, BrentKung.new],
      value: BrentKung.new);
  print('knob= ${prefixTreeKnob.toJson()}');
  print('choice0= ${prefixTreeKnob.choices[0].toString()}');
}
```
Output:
```
knob= {value: Closure: (List<Logic>, (Logic, Logic) => Logic) => BrentKung from Function 'BrentKung.': static.}
choice0= Closure: (List<Logic>, (Logic, Logic) => Logic) => Ripple from Function 'Ripple.': static.
```

Reply:
here is the full JSON of the Configurator:
```
JSON: {
  "name": "Parallel Prefix Adder",
  "knobs": {
    "Tree type": {
      "value": "Closure: (List<Logic>, (Logic, Logic) => Logic) => BrentKung from Function 'BrentKung.': static."
    },
    "Data width": {
      "value": 8
    }
  }
}
```

Reply:
haha yeah that won't look so great in a drop-down I suppose...

You could make a `Map<Type,Function>` which maps from things like `{BrentKung : BrentKung.new}` maybe so that the drop-down is nice, and then use the lookup to control building of the module?